internal/controller/serverset/serverset.go

```diff
@@ -167,16 +168,32 @@ func (e *external) populateReplicasStatuses(ctx context.Context, cr *v1alpha1.Se
 
 	for i := range serverSetReplicas {
 		replicaStatus := computeStatus(serverSetReplicas[i].Status.AtProvider.State)
+		errMsg := ""
+
+		lastCondition, err := getLastCondition(serverSetReplicas[i])
+		if err == nil && lastCondition.Reason == xpv1.ReasonReconcileError {
+			replicaStatus = statusError
+			errMsg = lastCondition.Message
+		}
+
 		cr.Status.AtProvider.ReplicaStatuses[i] = v1alpha1.ServerSetReplicaStatus{
 			Role:         fetchRole(ctx, e, serverSetReplicas[i]),
 			Name:         serverSetReplicas[i].Name,
 			Status:       replicaStatus,
-			ErrorMessage: "",
+			ErrorMessage: errMsg,
 			LastModified: metav1.Now(),
 		}
 	}
 }
 
+func getLastCondition(server v1alpha1.Server) (xpv1.Condition, error) {
```

Comment:
I don't think you need to return an error here, you can just return the condition and check the reason against xpv1.ReasonReconcileError anyway?